src/ServiceInsight/ServiceControl/DefaultServiceControl.cs

```diff
@@ -134,7 +141,16 @@ public IEnumerable<StoredMessage> GetConversationById(string conversationId)
 
         public IEnumerable<Endpoint> GetEndpoints()
         {
-            var request = new RestRequestWithCache(EndpointsEndpoint, RestRequestWithCache.CacheStyle.IfNotModified);
+            var rootUrls = GetRootUrls();
+
+            var endpointsUrl = rootUrls.KnownEndpointsUrl ?? rootUrls.EndpointsUrl ?? DefaultEndpointsEndpoint;
+
+            if (endpointsUrl.Contains(connection.Url))
+            {
+                endpointsUrl = endpointsUrl.Replace(connection.Url, string.Empty);
```

Comment:
could we just use replace without the contains?

Reply:
As discussed: yes.